### PR TITLE
Store certificate and private key in encrypted CosmosDB database

### PR DESCRIPTION
**Pull request by zkokelj**
### Why this change is needed

We want to have static https certificate (using the same private key) even after restarts & certificate renewals etc.
Certificate and private key were previously stored in memory and in file on attached volume. 
This is not ideal as it is stored unencrypted and in case of volume is deleted the private key and certificate are deleted.

Much better solution is to store both private key and certificate in encrypted CosmosDB.

### What changes were made as part of this PR

- implementation of `autocert.Cache` in which we store certificate and private key in cosmosDB and encrypt it.
- using this approach only for CosmosDB (for other databases - sqlite, we still use file storage for certificate and private key)
- 

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: tools/walletextension/storage/database/cosmosdb/cert_storage_cosmos.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	_, err := c.certsContainer.DeleteItem(ctx, partitionKey, keyString, nil)
+	if err != nil && !strings.Contains(err.Error(), "NotFound") {
+		return fmt.Errorf("failed to delete certificate: %w", err)
+	}
+
+	return nil
+}
+
+// dbKey generates a consistent key for CosmosDB storage
+func (c *CertStorageCosmosDB) dbKey(key []byte) (string, azcosmos.PartitionKey) {
```
Comment (tudor-malene) [inline tools/walletextension/storage/database/cosmosdb/cert_storage_cosmos.go:125]: I don't think we need to obfuscate the key. There will be only one

**Review by tudor-malene — COMMENTED** [review]
lgtm.
minor comment

File: tools/walletextension/storage/database/cosmosdb/cert_storage_cosmos.go (hunk 1 above)
Comment (zkokelj) [reply tools/walletextension/storage/database/cosmosdb/cert_storage_cosmos.go:124]: image

First 2 are old (obfuscated keys) and the last two are keys after the change.

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**